openid-connect-server/src/main/java/org/mitre/openid/connect/web/WhitelistAPI.java

```diff
@@ -167,6 +180,14 @@ public String updateWhitelistedSite(@PathVariable("id") Long id, @RequestBody St
 		}
 	}
 
+	private void validateWhitelistScopes(Set<String> scopes) throws ScopeException {
+		for (String s : scopes) {
+			if (!s.matches(characterMatcher)) {
```

Comment:
In class ScopeAPI and here. Looks like we could be doing regex matching multiple times, maybe we can create the Pattern object ahead of time for some performance benefit.

```
private static final Pattern pattern = Pattern.compile(characterMatcher);
....
private void validateWhitelistScopes(Set<String> scopes) throws ScopeException {
    for (String s : scopes) {
        if (!pattern.matcher(s).matches()) {
            throw new ScopeException(s);
        }
    }
}
```

Reply:
Good shout, adjusted